packages/@uppy/core/src/Uppy.ts

```diff
@@ -89,7 +89,7 @@ interface Plugins extends Record<string, Record<string, unknown> | undefined> {}
 
 export interface State<M extends Meta, B extends Body>
   extends Record<string, unknown> {
-  meta: M
+  meta: Partial<M>
```

Comment:
I think this is incorrect and whether it's `Partial` or not is up to the implementer of the generic.

Reply:
Hm, so state is supposed to have all metadata from M (e.g. not partial)? then we need to either:
- change UppyOptions so that it also takes in `meta: M` instead of `meta: Partial<M>`
- or change this line https://github.com/transloadit/uppy/blob/e4daea99ad41cfd14da1febb79f10a4ec2aa8789/packages/%40uppy/core/src/Uppy.ts#L388 so that it somehow sets default values for all M keys, but I don't know how to do that because M is not known

Reply:
> so state is supposed to have all metadata from M

Yes that's correct

Reply:
i've now tried to update the meta types to `M` instead of `Partial<M>` but TS is failing because we are trying to assign this.opts.meta to `Partial<M>` (`{}`) here:

https://github.com/transloadit/uppy/blob/e4daea99ad41cfd14da1febb79f10a4ec2aa8789/packages/%40uppy/core/src/Uppy.ts#L352

I think it would be a breaking change to alter this behavior

Reply:
We can do this as a type `meta: M | undefined`. Which is different from making all properties optional.

Reply:
I don't think that would help because {} is not assignable to `M | undefined`. or do you mean changing the runtime behavior to:
```
meta: opts?.meta
```

Reply:
if i do that i get
```
M could be instantiated with an arbitrary type which could be unrelated to M | undefined .
```